Dockerfile

```diff
@@ -106,6 +106,13 @@ RUN pip install --upgrade \
 RUN echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list \
  && wget http://packages.osrfoundation.org/gazebo.key -O - | apt-key add -
 
+# optionally add packages.ros.org as a source
+ARG USE_OSRF_REPOS
+RUN if [ "${USE_OSRF_REPOS}" = "True" ]; then \
+         echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -cs) main" > /etc/apt/sources.list.d/ros-latest.list \
+      && apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
```

Comment:
the keyserver has been known to be really unreliable.

Perhaps we should consider importing the key directly from `http://packages.ros.org/ros.key` as we do with `gazebo.key`.